cmd/fs-v1.go

```diff
@@ -656,17 +656,19 @@ func (fs *FSObjects) getObjectInfo(ctx context.Context, bucket, object string) (
 	// parallel Put() operations.
 
 	rlk, err := fs.rwPool.Open(fsMetaPath)
+	var n int64
 	if err == nil {
 		// Read from fs metadata only if it exists.
-		_, rerr := fsMeta.ReadFrom(ctx, rlk.LockedFile)
+		var rerr error
+		n, rerr = fsMeta.ReadFrom(ctx, rlk.LockedFile)
```

Comment:
We can simply ignore any error other than `if rerr != nil && rerr != io.EOF {`  and io.EOF can be considered as errFileNotFound. 

BTW the original problem in the issue is not that we don't handle io.EOF but to figure out why it is giving io.EOF in the first place, since a successfully written object with `fs.json` in-tact, should never give io.EOF i.e it shouldn't be empty.